etl/transform/elasticsearch.py

```diff
@@ -13,173 +18,152 @@
 
 
-def index_batch(tmp_index_dir, batch):
-    path = get_folder_path([tmp_index_dir, str(random.random())], recreate=True)
-    store = IndexStore(path)
-    for data in batch:
-        store.dump(data)
-    return store.get_index_by_id()
+def generate_elasticsearch_document(options):
+    document_type, document_transform, uri_data_index, data_json = options
+    document = json.loads(data_json)
+    if document_transform:
+        resolved = remove_empty(resolve(document_transform, document, uri_data_index))
+        document.update(resolved)
+    return document_type, document
 
 
-def index_on_disk(tmp_index_dir, data_list, pool, logger):
-    logger.debug("Indexing data on disk...")
+def generate_elasticsearch_documents(restricted_documents, document_configs_by_entity,
+                                     uri_data_index: UriIndex, pool: Pool, logger: Logger):
+    """
+    Produces and iterable of tuples (of document type and document) generated using the
+    document templates in configuration.
+    """
+    logger.debug("Preparing documents generation...")
 
-    index = partial(index_batch, tmp_index_dir)
+    # Prepare list of args for the 'generate_elasticsearch_document' function to run in a thread pool
+    def list_documents():
+        """
+        List data from index for which we have a document to generate
+        """
+        for entity_name in uri_data_index.keys():
+            uri_index = uri_data_index[entity_name]
+            for data_json in uri_index.values():
+                document_configs = document_configs_by_entity.get(entity_name) or [{}]
+                for document_config in document_configs:
+                    document_type = document_config.get('document-type') or entity_name
+                    if restricted_documents and document_type not in restricted_documents:
+                        continue
+                    document_transform = document_config.get('document-transform')
+                    yield document_type, document_transform, uri_data_index, data_json
+                if is_checkpoint(document_count):
+                    uri_index.close()
+                    uri_index.open()
+            uri_index.close()
 
-    batches = split_every(1000, data_list)
-    index_list = pool.imap_unordered(index, batches)
+    logger.debug("Generating documents...")
+    document_tuples = pool.imap_unordered(
+        generate_elasticsearch_document,
+        list_documents(),
+        CHUNK_SIZE
+    )
 
-    global_index = reduce(DataIdIndex.merge, index_list, DataIdIndex({}))
-    return global_index
+    document_count = 0
+    for document_tuple in document_tuples:
+        document_count += 1
+        yield dict(document_tuple)
+    logger.debug(f"Generated {document_count} documents.")
 
 
-def generate_elasticsearch_document(options):
-    document_type, document_transform, data_id, data_index = options
-    document = data_index[data_id]
+def generate_elasticsearch_document2(options):
```

Comment:
generate_elasticsearch_documents seems to be deprecated and replaced with generate_elasticsearch_document2
COnsider removing generate_elasticsearch_documents and rename generate_elasticsearch_document2 to generate_elasticsearch_documents